ros/angel_system_nodes/angel_system_nodes/intent_to_command.py

```diff
@@ -0,0 +1,134 @@
+import rclpy
+from rclpy.node import Node
+import yaml
+
+from angel_msgs.msg import (
+    InterpretedAudioUserIntent,
+    SystemCommands,
+)
+
+
+class IntentToCommand(Node):
+    """
+    ROS node that converts user intent messages to system commands.
+    """
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        self._explicit_intent_topic = (
+            self.declare_parameter("explicit_intent_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self._confirmed_intent_topic = (
+            self.declare_parameter("confirmed_intent_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self._sys_cmd_topic = (
+            self.declare_parameter("sys_cmd_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self._intent_to_cmd_map_path = (
+            self.declare_parameter("intent_to_cmd_map", "")
+            .get_parameter_value()
+            .string_value
+        )
+
+        if self._explicit_intent_topic == "":
+            raise ValueError(
+                "Please provide the explicit intent topic with the `explicit_intent_topic` parameter"
+            )
+        if self._confirmed_intent_topic == "":
+            raise ValueError(
+                "Please provide the confirmed intent topic with the `confirmed_intent_topic` parameter"
+            )
+        if self._intent_to_cmd_map_path == "":
+            raise ValueError(
+                "Please provide the intent command map path with the `intent_to_cmd_map` parameter"
+            )
+        if self._sys_cmd_topic == "":
+            raise ValueError(
+                "Please provide the system command topic with the `sys_cmd_topic` parameter"
+            )
+
+        log = self.get_logger()
+        log.info(f"Explicit intent topic: {self._explicit_intent_topic}")
+        log.info(f"Confirmed intent topic: {self._confirmed_intent_topic}")
+        log.info(f"System command topic: {self._sys_cmd_topic}")
+        log.info(f"Intent to cmd map: {self._intent_to_cmd_map_path}")
+
+        # Load intent to cmd map (yaml)
+        with open(self._intent_to_cmd_map_path, 'r') as f:
+            config = yaml.safe_load(f)
+        self._intent_to_cmd_map = config["commands"]
+
+        self._subscription = self.create_subscription(
+            InterpretedAudioUserIntent,
+            self._explicit_intent_topic,
+            self.intent_callback,
+            1
+        )
+        self._subscription = self.create_subscription(
+            InterpretedAudioUserIntent,
+            self._confirmed_intent_topic,
+            self.intent_callback,
+            1
+        )
```

Comment:
Maybe these topics don't need to be separated. We could probably just have one topic that inputs to this that both the "explicit" and "confirmed" sources publish to.